#!/usr/bin/env python

# set up parameters that we care about
PACKAGE = 'jsk_pcl_ros'

from dynamic_reconfigure.parameter_generator_catkin import *;

from math import pi

gen = ParameterGenerator ()
gen.add("magnify_distance", double_t, 0, "Lenght to magnify convex polygons", 0.2, 0.0, 1.0)
gen.add("distance_threshold", double_t, 0, "Allowed distance threshold from plane to points", 0.01, 0.0, 1.0)
gen.add("normal_threshold", double_t, 0, "Allowed normal threshold from plane to points", 0.3, 0.0, pi)
gen.add("resolution", double_t, 0, "Resolution of grid", 0.005, 0.001, 0.1)
gen.add("morphological_filter_size", int_t, 0, "Size of morphological filtering (dilation and erosion)", 1, 0, 10)
gen.add("erode_filter_size", int_t, 0, "Size of erode filtering", 1, 0, 10)
gen.add("footprint_plane_distance_threshold", double_t, 0, "Distance threshold to lookup footprint plane", 0.01, 0.0, 1)
gen.add("footprint_plane_angular_threshold", double_t, 0, "Angular threshold to lookup footprint plane", 0.1, 0.0, pi/2)


exit (gen.generate (PACKAGE, "jsk_pcl_ros", "EnvironmentPlaneModeling"))
